scripts/create_role_molecule.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import yaml
+from jinja2 import Environment, FileSystemLoader
+
+
+PROJECT_DIR = Path(__file__).resolve().parent.parent
+SCRIPT_DIR = PROJECT_DIR / 'scripts'
+CI_TEMPLATES_DIR = PROJECT_DIR / 'ci' / 'templates'
+CI_CONFIG_DIR = PROJECT_DIR / 'ci' / 'config'
+ROLES_DIR = PROJECT_DIR / 'ci_framework' / 'roles'
+ZUUL_JOB_DIR = PROJECT_DIR / 'zuul.d'
+SOURCE_ROLES_DOCS_DIR = PROJECT_DIR / 'docs' / 'source' / 'roles'
+
+
+def generate_zuul_job_role_yaml(roles_name):
+    """
+    Generate YAML structure for a Zuul job based on given role.
+
+    Args:
+        roles_name (str): Name of the role.
+
+    Returns:
+        dict: YAML structure for the role.
+    """
+    template_loader = FileSystemLoader(searchpath=CI_TEMPLATES_DIR)
+    template_env = Environment(loader=template_loader)
+    template = template_env.get_template('molecule.yaml.j2')
+    zuul_rendered_template_jobs = yaml.safe_load(template.render(role_names=roles_name))
+    return zuul_rendered_template_jobs
+
+
+def regenerate_projects_zuul_jobs_yaml():
+    """
+    Regenerate zuul.d/projects.yaml file with the name of each role
+    under ci-framework/roles
+    """
+    project_yaml_path = CI_TEMPLATES_DIR / 'projects.yaml'
```

Comment:
If it's constant can be perfectly moved to the top of the file with the other "constants" https://github.com/openstack-k8s-operators/ci-framework/pull/340/files#diff-5d3878a84b579ecca7e889822946d5e58c6df4c4ef136f7646baea74e85d8ec7R10-R14

Reply:
in addition... I'd use `os.path.join()`